src/server/game/Entities/Player/Player.cpp

```diff
@@ -7309,6 +7299,31 @@ void Player::SetArenaPoints(uint32 value)
         AddKnownCurrency(ITEM_ARENA_POINTS_ID);
 }
 
+void Player::SendPVPCreditMessage(int32 honor, ObjectGuid victimGuid, uint32 victimRank)
+{
+    // Sends honor message to client (chat log, floating text)
+    
+    // Packet details:
+    //      honor - amount of honor to show in log
+    //      victim_guid - to show victim name in log
+    //      victim_rank [1..4]  HK: <dishonored rank>
+    //      victim_rank [5..19] HK: <alliance\horde rank>
+    //      victim_rank [0, 20+] HK: <>
+    //
+    //      NOTE: The "HK:" is also displayed as floating text
+
+    // Defaulting victim_guid and victim_rank to zero prevents the client from displaying an
+    //  empty "HK:" floating message if victim has no rank, such as an NPC (script: player_creature_honor)
+    //  or an invalid player (though the second case should NEVER happen)
+
+    WorldPacket data(SMSG_PVP_CREDIT, 4 + 8 + 4);
+    data << honor;
+    data << uint64((victimGuid.IsPlayer() ? victimGuid.GetCounter() : 0)); // must force uint64 or results in wrong message
```

Comment:
> WTF is this? This field is a GUID so why are you only sending a part of it?

Because GetCounter() in the 335 branch sends the GUID as it's found in the database. The other way didn't work.